client/executor/src/deprecated_host_interface.rs

```diff
@@ -406,7 +448,10 @@ impl_wasm_host_interface! {
 		) -> Pointer<u8> {
 			let storage_key = context.read_memory(storage_key_data, storage_key_len)
 				.map_err(|_| "Invalid attempt to determine storage_key in ext_child_storage_root")?;
-			let value = runtime_io::storage::child_root(&storage_key);
+			// This can only work with archive mode.
+			let value = runtime_io::storage::child_root(
+				&storage_key,
+			);
```

Comment:
can this only works with archive mode ? I thought that now runtime_io::storage::child_root was fixed.

Reply:
That is the 'deprecated_host_interface' where I kept old behavior.
In fact this should be deleted since not use in kusama, but I was not bold enough to do it (that is the first bold point in description).
I think I shall remove it, anyway I will test kusama synch again (last test was a bit to far away and with parasitic trie behavior).
So idea would be to remove all child related call from deprecated host, agree?